Legion/Various/UndeadLegionMerge.cs

```diff
@@ -0,0 +1,212 @@
+    // [Can Change] This should only be changed by the author.
+    //              If true, it will not stop the script if the default case triggers and the user chose to only get mats
+    private bool dontStopMissingIng = false;
+
+    public void ScriptMain(IScriptInterface bot)
+    {
+        Core.BankingBlackList.AddRange(new[] { "Legion Token", "Frosted Falchion", "Judgement Scythe", "Judgement Hammer", "Cursed Scimitar", "Essence of the Undead Legend", "Shadow Shroud", "DragonBlade of Nulgath", "Fallen MonsterHunter", "Fallen MonsterHunter Helm", "Fallen MonsterHunter Cape", "Fallen MonsterHunter Sword", "Exalted Crown " });
+        Core.SetOptions();
+
+        BuyAllMerge();
+
+        Core.SetOptions(false);
+    }
+
+    public void BuyAllMerge()
+    {
+        //Only edit the map and shopID here
+        Adv.StartBuyAllMerge("underworld", 238, findIngredients);
+
+        #region Dont edit this part
+        void findIngredients()
+        {
+            ItemBase req = Adv.externalItem;
+            int quant = Adv.externalQuant;
+            int currentQuant = req.Temp ? Bot.TempInv.GetQuantity(req.Name) : Bot.Inventory.GetQuantity(req.Name);
+            if (req == null)
+            {
+                Core.Logger("req is NULL");
+                return;
+            }
+
+            switch (req.Name)
+            {
+                default:
+                    bool shouldStop = Adv.matsOnly ? !dontStopMissingIng : true;
+                    Core.Logger($"The bot hasn't been taught how to get {req.Name}." + (shouldStop ? " Please report the issue." : " Skipping"), messageBox: shouldStop, stopBot: shouldStop);
+                    break;
+                #endregion
+
+                case "Legion Token":
+                    Legion.FarmLegionToken(quant);
+                    break;
+
+                case "Frosted Falchion":
+                    Core.FarmingLogger(req.Name, quant);
+                    Adv.BuyItem("BlindingSnow", 236, req.Name);
+                    break;
+
+                case "Judgement Scythe":
+                    LegionExercise4.Exercise(new[] { "Judgement Scythe", "Legion Token" });
+                    break;
+
+                case "Judgement Hammer":
+                    LegionExercise3.Exercise(new[] { "Judgement Hammer", "Legion Token" });
+                    break;
+
+                case "Cursed Scimitar":
+                    Core.FarmingLogger(req.Name, quant);
+                    Adv.BuyItem("SandSea", 242, req.Name);
+                    break;
+
+                case "Essence of the Undead Legend":
+                    Core.Logger($"{req.Name} Is seasonal item from Dage's Dark Birthday Shop");
+                    Core.FarmingLogger(req.Name, quant);
+                    Adv.BuyItem("DarkBirthday", 376, req.Name);
+                    break;
+
+                case "Shadow Shroud":
+                    if (!Core.CheckInventory(req.Name, quant))
```

Comment:
There is already an internal check, if the bot even comes to this case its cuz it doesnt have the item